### ISSUE TITLE
Presence of zowe Explorer stops VSCode F5 debugging key from working in another extension

### ISSUE THREAD

**Issue by davidkjackson54** [zowe/vscode-extension-for-zowe#2285]
I have **zowe Explorer V2.8.0** installed. 
I am developing my own VSCode Explorer extension which for the past 2 years has worked fine and I am able to debug it using the F5 command to start debugging and then continuous F5 to step through breakpoints. 
Now however, I can click F5 to start debugging but then repeated F5 has no effect and instead I have to click Run-Continue 
 to step through Breakpoints. This started about 2 weeks ago.

 I opened a VSCODE issue for this - [https://github.com/microsoft/vscode/issues/182658]
I ran their suggestion and this is the output which seems to suggest that **zowe Explorer** is interfering with the F5
 
```
2023-05-17 06:23:36.103 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-05-17 06:23:36.145 [info] [KeybindingService]: | Resolving F5
2023-05-17 06:23:36.145 [info] [KeybindingService]: \ From 4 keybinding entries, matched zowe.manualPoll, when: editorIsOpen, source: user extension Zowe.vscode-extension-for-zowe.
2023-05-17 06:23:36.146 [info] [KeybindingService]: / Received  keydown event - modifiers: [], code: F5, keyCode: 116, key: F5
2023-05-17 06:23:36.146 [info] [KeybindingService]: | Converted keydown event - modifiers: [], code: F5, keyCode: 63 ('F5')
2023-05-17 06:23:36.146 [info] [KeybindingService]: | Resolving F5
2023-05-17 06:23:36.146 [info] [KeybindingService]: \ From 4 keybinding entries, matched zowe.manualPoll, when: editorIsOpen, source: user extension Zowe.vscode-extension-for-zowe.
2023-05-17 06:23:36.147 [info] [KeybindingService]: + Invoking command zowe.manualPoll.
```
I then disabled zowe Explorer and then ran my extension in debug and F5 then started to work as it has previously without any issue.

Re-enabling zowe Explorer and the F5 again stopped working

**Comment by davidkjackson54** [comment]
Although I have the zowe Explorer installed, it was not in use at the time. I was debugging my own extension.
**Yes, changing that value does then allow my extension to be debugged.**
Why, when zowe Explorer is not running - interfere with other extensions?
What version of zowe Explorer introduced this bug?

**Comment by adam-wolfe** [comment]
This was introduced in v2.8.0 to give users a key to refresh spool files. As it is currently implemented, Zowe Explorer will override actions bound to the F5 key when an editor tab is open.

@traeok, could we use resourceDirname to limit the context for the keybind?

**Comment by traeok** [comment]
> @traeok, could we use resourceDirname to limit the context for the keybind?

I found that we could use `resourceScheme` so that the key binding only activates when the editor has a spool resource open (`zosspool` scheme). Addressed in commit 59067fc, will be released with next patch fix for Zowe Explorer.

**Comment by davidkjackson54** [comment]
Thanks for the quick action on this one.

**Comment by zFernand0** [comment]
> > @traeok, could we use resourceDirname to limit the context for the keybind?
> 
> I found that we could use `resourceScheme` so that the key binding only activates when the editor has a spool resource open (`zosspool` scheme). Addressed in commit [59067fc](https://github.com/zowe/vscode-extension-for-zowe/commit/59067fcbc7e55705e5b50fa338e0c8f2eed59456), will be released with next patch fix for Zowe Explorer.

Should we add some developer documentation around keybinds?
This may already be a problem with some of the other keys 😋 

Ideally we will have all keybinds associated with a resourceScheme 😋

**Comment by davidkjackson54** [comment]
I have come across another instance where Zowe Explorer is interferingvwith VSCODE 
Shift Z is being intercepted by zowe - so Iam unable to enter any upper case Z..
![image](https://github.com/zowe/vscode-extension-for-zowe/assets/46171060/42cdabf4-8563-42e4-aef8-24fc5c14fa66)

**Comment by traeok** [comment]
> I have come across another instance where Zowe Explorer is interferingvwith VSCODE Shift Z is being intercepted by zowe - so Iam unable to enter any upper case Z.. ![image](https://user-images.githubusercontent.com/46171060/245954804-42cdabf4-8563-42e4-aef8-24fc5c14fa66.png)

Hi @davidkjackson54 ,

Would you mind confirming if there are two entries for "Zowe Explorer: Poll Content in Active Editor" in Settings -> "Keyboard Shortcuts"?

You might be running into a side effect of changing the keybinding before the 2.8.1 update. Since VS Code separates user bindings based on their context values, it might have saved the old context value `editorIsOpen` when the keybinding was changed. See below for an example for one of the system shortcuts:

![image](https://github.com/zowe/vscode-extension-for-zowe/assets/86500639/096c58ff-e3ab-4211-991d-caf62fcb9549)

If this is the case, removing your assigned keybinding for "Zowe Explorer: Poll Content in Active Editor", and then reassigning the shortcut should resolve the issue. This should update your shortcut to use the corrected context value included w/ versions 2.8.1 and above.

If this is not the case, we can reopen this issue (or make a follow-up issue) to resolve this as soon as possible.

Thanks!

**Comment by davidkjackson54** [comment]
Yes there was . I guess it was present as a workaround that you gave me for the original problem. However, it is strange that this only started to occur this week. After all Shift Z is commonly used here.
![image](https://github.com/zowe/vscode-extension-for-zowe/assets/46171060/8e7e71d6-5124-48ab-88dd-1d00451dd9f8)
